src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -2307,7 +2310,15 @@ NetworkOPsImp::recvValidation(
     JLOG(m_journal.trace())
         << "recvValidation " << val->getLedgerHash() << " from " << source;
 
+    std::unique_lock lock(validationsMutex_);
+    if (pendingValidations_.contains(val->getLedgerHash()))
+        return false;
+    pendingValidations_.insert(val->getLedgerHash());
+    lock.unlock();
     handleNewValidation(app_, val, source);
```

Comment:
@mtrippled, my understanding is this is meant to prevent later on the calls to [`checkAccept`](
https://github.com/ximinez/rippled/blob/098546f0b1ba259d144e3e60fc9b3274f85e5d05/src/ripple/app/consensus/RCLValidations.cpp#L189) in`handleNewValidation` that may end up calling starting of a chain of `InboundLedger` requests here https://github.com/ximinez/rippled/blob/098546f0b1ba259d144e3e60fc9b3274f85e5d05/src/ripple/app/ledger/impl/LedgerMaster.cpp#L1070-L1071. 

Asking to understand if you instead considered change the call at this `LedgerMaster` callsite to use the new `acquireAsync` method you added, and if so, why you went this way? I'm guessing you are trying to filter out redundant validations at the highest level?